eng/packages/http-client-csharp/generator/Azure.Generator/src/AzureArmVisitor.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Generator.CSharp.ClientModel;
+using Microsoft.Generator.CSharp.ClientModel.Providers;
+using Microsoft.Generator.CSharp.Primitives;
+using Microsoft.Generator.CSharp.Providers;
+using System.IO;
+
+namespace Azure.Generator
+{
+    // only apply for MPG
+    internal class AzureArmVisitor : ScmLibraryVisitor
+    {
+        /// <inheritdoc/>
+        protected override TypeProvider? Visit(TypeProvider type)
+        {
+            if (type is ClientProvider)
+            {
+                type.Update(modifiers: TransfromModifiers(type), relativeFilePath: TransformRelativeFilePathForClient(type));
+            }
+            //if (type is RestClientProvider)
```

Comment:
If we make both client and restclient internal, all the models referenced will be internalized during the postprocess.
Once we have the resource and collection in place, public reference will be in place and the referenced models will be pubilc.
So, keep restclient public for now to make the referenced models public.